### DISCUSSION TITLE
Why flutter is being loved more than UNO?

### DISCUSSION THREAD

**Discussion by fisforfaheem** [unoplatform/uno#4980] [General]
Why flutter seems better?? also why cant you make great fluent UI apps for android or mac?? i mean porting a bare calculator apps wont make u. a star u need a flutter like community and ease of building apps

**Comment by IsmailHassani** [comment]
I believe that Flutter and Uno serve different audiences. From a multi-platform perspective, they both do the same, especially now Uno supports WPF and Linux. 
However, as a C# (.Net) developer, for me the transition from Windows Forms > Silverlight > WPF > UWP to Uno feels more natural than Flutter. 
On the other hand, i guess that for webdevelopers, iOS and Android developers the transition to Flutter is more logical.
So why flutter is being more loved depends solely on whom you ask.

Before Uno we (C#/XAML developers) had only Xamarin with a fairly limited platform support. Now we can reuse mostly 95-98% of our code (including UI) in Uno to deliver the same experience to multiple platforms.
Of course, there is still long way to go to be real plug and play, but the difference between now and 2 years ago is epic.

I, myself, am in process to transition a big LOB application to Uno. 
I have also to fight the challenges that a new platform gives you in terms of compatibility, controls and also the fast-moving pace of different components one uses in such application. 
Despite the lack of good examples of applications that are converted to Uno, i can now publish my application as UWP, Android, iOS, Linux, WPF and Web Assembly. I'm not there yet, but I'm starting to like this journey more and more.
If you want, you can check my progress on my [blog](https://ismail.hassani.eu/posts/).

**Comment by realtebo** [comment]
Uno has not a full video course available. No one. Search on udemy. No one a single course. How do you think you attract new developers? I am actively looking for books and video courses, paid ..., And cannot find nothing

**Reply by sasakrsmanovic** [reply]
Hello. There are numerous video courses available, even at outlets you specify
https://www.pluralsight.com/courses/uno-platform-fundamentals
https://www.udemy.com/course/introduction-to-uno-platform/  (A bit older)

Our own Uno Tech Bites with some 40 videos to date, plus we publish about 2 per week. 
https://www.youtube.com/playlist?list=PLl_OlDcUya9rP_fDcFrHWV3DuP7KhQKRA

I hope that gives you some materials to explore. Thanks for considering Uno Platform!

**Comment by pkar70** [comment]
But you have plenty videos how to use UWP!
Uno is not new programming platform, Uno is "only" way to run UWP programs on platforms other than Windows. Ideally, you take your UWP app, add Uno as Nuget, and voile'a - you can build same app for iOS, macOS, Linux, Android and any other platform.

**Comment by IsmailHassani** [comment]
I totally agree with you that the lack of books and/or videos indeed deters many potentially new Uno developers. I think that is also why many people are not yet hooked up. 
Don't forget that this is a very young platform that has grown very fast in a very short time. As a result, a number of "obvious" sources are simply not yet available or sufficient. The matter is also subject to many fast changes.

One thing I'm impressed by is the amazing and great online community that is growing by the day. And the help and willingness of these people has made me decide to take the plunge anyway.
The lack of documentation led me to write a blog in early September and try to contribute to the community. 
So don't let this stop you from taking a chance. 
The learning curve is still steep, but as you get started it's not so bad. 

The alternative is to have to use another platform that doesn't already have Uno's features (Flutter, Avalonia or Xamarin) or simply forget multi-platform apps. All valid choices. 
But I love learning new things myself. I especially love bleeding-edge tech. Sometimes it's been for nothing (like Windows Phone) and otherwise it's a springboard to something new (Silverlight > WPF > UWP > ???).

So don't be afraid, if you have any questions ask them here or in Discord and people will help you as much they can.
;)

**Reply by sasakrsmanovic** [reply]
Thanks, for the note on community; we try to be as welcoming as possible and it is great to see you are recognizing that. 

Also  I just want to make sure you are aware of all the books written on Uno Platform:
https://www.packtpub.com/product/creating-cross-platform-c-applications-with-uno-platform/9781801078498
https://www.oreilly.com/library/view/getting-started-with/9781484282489/html/512245_1_En_1_Chapter.xhtml
https://www.syncfusion.com/succinctly-free-ebooks/uno-platform-succinctly

In addition, Uno Platform has sections at another few books authored by @alvinashcraft 

I answered the note on videos above, but here it is again: 
https://www.pluralsight.com/courses/uno-platform-fundamentals
https://www.udemy.com/course/introduction-to-uno-platform/ (A bit older)

Our own Uno Tech Bites with some 40 videos to date, plus we publish about 2 per week.
https://www.youtube.com/playlist?list=PLl_OlDcUya9rP_fDcFrHWV3DuP7KhQKRA

**Comment by realtebo** [comment]
I find only now the first video on udemy thanks to customer care

I am very interested. But I need a from scratch course.  But from zero to desktop app . Without firebase . I need SQLite or morr.

It is really hard find courses about the platform.

You all known all pieces needed around Uno. We no. We need a whole figure introduction.

No intro.? No new devs.

I will buy the udemy course.

Ottieni Outlook per Android<https://aka.ms/ghei36>

________________________________

**Reply by sasakrsmanovic** [reply]
I love to hear that you are going to try with the Udemy course. It is free btw, but it is also a bit dated. We have some more resources listed out at this blog
https://platform.uno/blog/uno-platform-learning-resources-2023-update/

But for learning you are best of with these step by step tutorials
https://platform.uno/simple-calc/ 
https://platform.uno/tube-player/

**Comment by IsmailHassani** [comment]
@realtebo I guess this is a whole other question. Learning UWP has nothing to do with Uno. However there are a lot sources how to create an UWP application.
[Get started with Windows 10 apps](https://docs.microsoft.com/en-us/windows/uwp/get-started/)
[Universal Windows Platform documentation](https://docs.microsoft.com/en-us/windows/uwp/)
[Several videos](https://www.bing.com/videos/search?q=uwp+tutorials&&view=detail&mid=A5E302FD8E01C1B985B6A5E302FD8E01C1B985B6&&FORM=VRDGAR&ru=%2Fvideos%2Fsearch%3Fq%3Duwp%2Btutorials%26qpvt%3Duwp%2Btutorials%26FORM%3DVDRE)

**Comment by pkar70** [comment]
@IsmailHassani: About year ago, I tried to make more "from scratch" guides, e.g. how to convert existing app to Uno.
Yes, some parts of Uno, especially when you try to contribute to it, is very hard and has poor documentation.

But I'm sure starting using Uno is simple and well documented - I converted many apps from VB/UWP to C#/UWP/Uno without problems.

**Comment by IsmailHassani** [comment]
@pkar70 I agree totally with you. But after last summer (UnoConf 2020) i would try it for the last time. I told myself this would be the last of total 3 attempts. Luckily I tried, because after this update, the experience was 180 degrees different. With the help of the community, it went just upwards. ;)

**Comment by realtebo** [comment]
@IsmailHassani  imho no, is not a whole other question.

See flutter:  you can EASILY find tutorial to
- learn ALL single steps to start coding from a clean empty new windows or mac pc
- learn a minimal amount of dart
- learn how to use flutter widgets
- learn how to "build" for each platforms


This is USEFULL


Compare with uno doc

Start here: https://platform.uno/
I click on getting started

And find this

"BEFORE YOU START THE TUTORIAL
If you haven't prepared your dev environment for Uno Platform app development yet, the Setting up the Environment page will guide you through installing all the prerequisites."

Ok, I click over this because i need to know EACH SINGLE PIECE TO INSTALL

I am now here: https://platform.uno/docs/articles/get-started.html

I choosed vs code 'for habit'.
https://platform.uno/docs/articles/get-started.html#vs-code

Instructions

```
Create an Uno Platform project
Launch Code, then in the terminal type the following to install the Uno Platform templates:
dotnet new -i Uno.ProjectTemplates.Dotnet
```

I see a discrete long output and try next command

```
dotnet new unoapp -o MyApp -ios=false -android=false -macos=false -uwp=false --vscodeWasm
```

Ok, then tutorial tell me 

"This will create a solution that only contains the WebAssembly platform support.

Prepare the WebAssembly application for debugging"

WHAT ?!?!!? 


No this is totally disordered. 
Learning curve is ONE of the problem of Uno.
Introduce gracefully WHAT IS, WHAT CAN DO, and drive a new develop step by step to prepare a windows 10 pc or a mac to start develop the classic hello world.

Than prepare step 2: some c#, some xaml, 
The step 3: build for windows 10 and build for mac
Then step 4: add android and mac

you can tell me to lean windows 10 app, then uwp then several videos, then bring back here.

HERE I need a bit of ALL of these. 





<blockquote><div><img src="https://platform.uno/app/themes/creatus-child/assets/img/favicon/favicon-16x16.png" height="14"> Uno Platform</div><div><strong><a href="https://platform.uno/">Uno Platform</a></strong></div><div>Pixel-Perfect Multi-Platform Applications with C# and WinUI for Windows, WebAssembly, iOS, macOS, Android and Linux</div></blockquote>
<blockquote><img src="../images/UnoLogoSmall.png" width="48" align="right"><div><strong><a href="https://platform.uno/docs/articles/get-started.html">Setting up your development environment</a></strong></div></blockquote>
<blockquote><img src="../images/UnoLogoSmall.png" width="48" align="right"><div><strong><a href="https://platform.uno/docs/articles/get-started.html#vs-code">Setting up your development environment</a></strong></div></blockquote>

**Comment by realtebo** [comment]
Second problem is mediatic.

I discovered uno platform only now. I cam from web dev. So it's obvious I didn't read 'specific' blogs about dotnet and c# and so on..

But the task to spread the word is totally your. write medium posts. write on dev.to. etc

I love c#, but it's confusing for me which of the 30 alternative to build a gui for windows is usable for cross-platform. I came here because of this problem: write a stupid app to save a 10 fields table on pc AND on mac. 

Add info about your project to blogs and forum and reddits where people talk about cross-platforms.

reply to cross-platform question on Stack overflow and speack of this alternative, briefly.

Marketing.

**Comment by realtebo** [comment]
For example: link books and online courses in a 'learning materials' section.

Litterally, I wrote to support email asking for videos or books. And NO BOOKS suggested me. 
I

**Comment by realtebo** [comment]
Oh....  after loosing my time to prepare visual studio code, ...

https://platform.uno/skill-and-code/

I discovered that: 

"Visual Studio (all target platforms), Visual Studio Code (WebAssembly target only)"

So i loose time. And must restart from visual studio setup. But... why enterprise ? not community? ... why?
No, i leave doc. I leave project.

<blockquote><img src="https://s3.amazonaws.com/uno-website-assets/wp-content/uploads/2020/08/05090629/ide-logos.png" width="48" align="right"><div><img src="https://platform.uno/app/themes/creatus-child/assets/img/favicon/favicon-16x16.png" height="14"> Uno Platform</div><div><strong><a href="https://platform.uno/skill-and-code/">Uno Platform Skill and Code</a></strong></div></blockquote>

**Comment by IsmailHassani** [comment]
@realtebo I feel you. But please understand what I told you earlier. The platform grows too fast to keep up. 
More importantly, documentation and especially manuals should be written by people who are didactically educated or experienced. In my career I have had to teach complete nitwits in the field of computers. I had to explain to then how Excel and Access work in a professional environment when they only saw cubes.
I don't blame them, but I had to teach them this somehow.
I also hope this comes up a bit in my blogs.

I think this is a logical future step in the development of Uno Platform to bring the documentation more educational than from the perspective of the programmer.

p.s. I'm not affiliated with Uno Platform in any way. I'm just a member of this community. But rest assured, knowing the guys at Uno, they're definitely going to do something with this.

**Comment by pkar70** [comment]
I don't know why not Community VStudio. I'm using community version, and this is sufficient also for contributing to Uno (adding functionality not currently implemented in Uno).

Did you try https://platform.uno/docs/articles/getting-started-tutorial-1.html ?

And... seems like previous version of migrating code from UWP to Uno was better :) But as you want new app, not migrate code, it is irrelevant.


<blockquote><img src="../images/UnoLogoSmall.png" width="48" align="right"><div><strong><a href="https://platform.uno/docs/articles/getting-started-tutorial-1.html">Create a Single Page App with Uno Platform</a></strong></div></blockquote>

**Comment by realtebo** [comment]
> I don't know why not Community VStudio. I'm using community version, and this is sufficient also for contributing to Uno (adding functionality not currently implemented in Uno).
> 
> Did you try https://platform.uno/docs/articles/getting-started-tutorial-1.html ?

It works, my fault. I read "somewhere" it requires enterprise.



<blockquote><img src="../images/UnoLogoSmall.png" width="48" align="right"><div><strong><a href="https://platform.uno/docs/articles/getting-started-tutorial-1.html">Create a Single Page App with Uno Platform</a></strong></div></blockquote>

**Comment by fisforfaheem** [comment]
I agree with you all ❤️. But being a person who has no idea real
programing, flutter seems very very easier and mature. Uno the company
itself should start making basic hello world programs and then add
videos... Flutter is only popular because it has videos but uno being
nothing different than uwp etc still needs its own channel like... Say
mvtechviral channel on YouTube.. or nick manning flutter channel etc please
🥺 don't ignore us.

**Comment by francoistanguay** [comment]
I'll close this issue as it is really a discussion. Feel free to discuss further in our Discussions.

**Comment by lindexi** [comment]
Now, part of the flutter team was laid off...

**Reply by francoistanguay** [reply]
We're still here and team is growing!

**Comment by fisforfaheem** [comment]
Flutter just seems so well designed, it's naming convention and widgets let
u build any thing...and u don't fee fragmented. As everything ons available
right in the sdk.

**Reply by sasakrsmanovic** [reply]
We cannot be sure as this is not a flutter repo :). Its best to ask this question at a flutter forum. 

But we do know why developers do chose us - and we documented that in this thread https://platform.uno/docs/articles/concepts/overview/why-uno-platform.html
